tests/testthat/helper-testing.R

```diff
@@ -17,30 +17,47 @@ rmot_test_single_individual <- function(model_name,
   data <- readRDS(test_path("fixtures", model_name,
                             paste0(model_name, "_data_single_ind.rds")))
 
-  # Test constant single individual
-  suppressWarnings( #Suppresses stan warnings
-    single_ind_test <- rmot_model(paste0(model_name, "_single_ind")) |>
-      rmot_assign_data(n_obs = data$n_obs, #integer
-                       y_obs = data$y_obs,
-                       obs_index = data$obs_index, #vector length N_obs
-                       time = data$time, #Vector length N_obs
-                       y_0_obs = data$y_0_obs #vector length N_ind
-      ) |>
-      rmot_run(chains = 1, iter = 1000, verbose = FALSE, show_messages = FALSE)
-  )
+  if(! is.null(data$step_size)){
+    # Test single individual
+    suppressWarnings( #Suppresses stan warnings
+      single_ind_test <- rmot_model(paste0(model_name, "_single_ind")) |>
+        rmot_assign_data(step_size = data$step_size,
+                         n_obs = data$n_obs, #integer
+                         y_obs = data$y_obs,
+                         obs_index = data$obs_index, #vector length N_obs
+                         time = data$time, #Vector length N_obs
+                         y_0_obs = data$y_0_obs #vector length N_ind
+        ) |>
+        rmot_run(chains = 1, iter = 1000, verbose = FALSE, show_messages = FALSE)
+    )
+  } else {
+    # Test single individual
+    suppressWarnings( #Suppresses stan warnings
+      single_ind_test <- rmot_model(paste0(model_name, "_single_ind")) |>
+        rmot_assign_data(n_obs = data$n_obs, #integer
+                         y_obs = data$y_obs,
+                         obs_index = data$obs_index, #vector length N_obs
+                         time = data$time, #Vector length N_obs
+                         y_0_obs = data$y_0_obs #vector length N_ind
+        ) |>
+        rmot_run(chains = 1, iter = 1000, verbose = FALSE, show_messages = FALSE)
+    )
+  }
 
   # Extract samples and check if parameter estimates are reasonable.
   ind_samples <- rstan::extract(single_ind_test, permuted = TRUE,
                                          inc_warmup = FALSE)
   par_ests <- c()
-  for(i in length(par_names)){
+  for(i in 1:length(par_names)){
```

Comment:
a safer option than `1:length(par_names)` is `seq_along(par_names)`. they differ when par_names has length 0.

Reply:
Done.